### ISSUE TITLE
[Phase 4] Model Evaluation - Metrics & SHAP Analysis

### ISSUE THREAD

**Issue by Sakeeb91** [Sakeeb91/genomic-variant-prediction#5]
## Summary

Evaluate model performance with comprehensive metrics and interpret predictions using SHAP for clinical trust.

**Parent Issue:** #1
**Depends On:** Phase 3 (Model Development)

## Objectives

- Compute PR-AUC and ROC-AUC metrics
- Generate calibration curves
- Perform SHAP feature importance analysis
- Analyze misclassified variants

## System Context

```
src/evaluation/
├── metrics.py          # PR-AUC, ROC-AUC
├── calibration.py      # Calibration curves
├── shap_analysis.py    # SHAP explanations
└── error_analysis.py   # Misclassification analysis
```

## Files to Create/Modify

| File | Action | Description |
|------|--------|-------------|
| `src/evaluation/metrics.py` | Create | Compute evaluation metrics |
| `src/evaluation/calibration.py` | Create | Plot calibration curves |
| `src/evaluation/shap_analysis.py` | Create | SHAP feature importance |
| `tests/test_evaluation.py` | Create | Evaluation tests |

### Metrics
- [ ] Compute ROC-AUC with confidence interval
- [ ] Compute PR-AUC (critical for imbalanced data)
- [ ] Generate precision-recall curves
- [ ] Report at multiple probability thresholds

### Calibration
- [ ] Plot reliability diagram
- [ ] Compute expected calibration error (ECE)
- [ ] Compare pre/post calibration
- [ ] Document calibration quality

### SHAP Analysis
- [ ] Compute SHAP values for test set
- [ ] Generate summary plot (feature importance)
- [ ] Generate dependence plots for top features
- [ ] Identify unexpected feature contributions

### Error Analysis
- [ ] Identify false positives and false negatives
- [ ] Analyze error patterns by gene/variant type
- [ ] Compare misclassified vs correctly classified
- [ ] Document common error modes

### PR-AUC with Bootstrap CI

```python
from sklearn.metrics import average_precision_score
import numpy as np

def pr_auc_with_ci(y_true, y_prob, n_bootstrap=1000):
    """Compute PR-AUC with bootstrap confidence interval."""
    observed = average_precision_score(y_true, y_prob)

    bootstrapped = []
    n = len(y_true)
    for _ in range(n_bootstrap):
        idx = np.random.choice(n, size=n, replace=True)
        score = average_precision_score(y_true[idx], y_prob[idx])
        bootstrapped.append(score)

    ci_lower = np.percentile(bootstrapped, 2.5)
    ci_upper = np.percentile(bootstrapped, 97.5)

    return {
        'pr_auc': observed,
        'ci_lower': ci_lower,
        'ci_upper': ci_upper
    }
```

### SHAP Analysis

```python
import shap
import matplotlib.pyplot as plt

def compute_shap_importance(model, X_test, max_samples=1000):
    """Compute SHAP values and generate plots."""
    # Subsample for speed
    if len(X_test) > max_samples:
        X_sample = X_test.sample(max_samples, random_state=42)
    else:
        X_sample = X_test

    # Compute SHAP values
    explainer = shap.TreeExplainer(model)
    shap_values = explainer.shap_values(X_sample)

    # Summary plot
    fig, ax = plt.subplots(figsize=(10, 8))
    shap.summary_plot(shap_values[1], X_sample, show=False)
    plt.tight_layout()

    return shap_values, fig
```

### Calibration Curve

```python
from sklearn.calibration import calibration_curve
import matplotlib.pyplot as plt

def plot_calibration(y_true, y_prob, n_bins=10):
    """Plot calibration curve."""
    prob_true, prob_pred = calibration_curve(y_true, y_prob, n_bins=n_bins)

    fig, ax = plt.subplots(figsize=(8, 8))
    ax.plot([0, 1], [0, 1], 'k--', label='Perfectly calibrated')
    ax.plot(prob_pred, prob_true, 's-', label='Model')
    ax.set_xlabel('Mean predicted probability')
    ax.set_ylabel('Fraction of positives')
    ax.set_title('Calibration Curve')
    ax.legend()

    # Expected calibration error
    ece = np.mean(np.abs(prob_true - prob_pred))

    return fig, ece
```

## Verification

```bash
# Generate evaluation report
python -c "
from src.evaluation.metrics import compute_metrics
from src.models.classifier import VariantClassifier
import pandas as pd

clf = VariantClassifier.load('models/variant_classifier.joblib')
df = pd.read_parquet('data/processed/features.parquet')
# ... split and evaluate
"

# SHAP analysis
python -m src.evaluation.shap_analysis --model models/variant_classifier.joblib --output reports/shap/

# Run tests
pytest tests/test_evaluation.py -v
```

## Technical Challenges

| Challenge | Mitigation |
|-----------|------------|
| SHAP slow on large data | Subsample to 1000 variants |
| PR-AUC more important | Report PR-AUC prominently |
| Calibration interpretation | Document expected vs observed |
| Error patterns complex | Focus on actionable insights |

## Definition of Done

- [ ] PR-AUC computed and exceeds 0.70
- [ ] ROC-AUC exceeds 0.85
- [ ] Calibration curve shows good calibration (ECE < 0.1)
- [ ] SHAP identifies top 10 contributing features
- [ ] Error analysis documented
- [ ] All tests pass with `pytest tests/test_evaluation.py`
